### PR TITLE
Create database

### PR DESCRIPTION
**Pull request by Stephen-Adom**
# Create a database based on a diagram
In this pull request, the following tasks were implemented:
- Created a file named `schema_based_on_diagram.sql` where you implement the database from the diagram.
- Added Join tables for many-to-many relationships
- Added the FK indexes
- Provided the link to the Entity relational diagram [Database diagram for clinic](https://lucid.app/lucidchart/7114a0c5-b394-4d9e-aa0d-b8b9bbf7506a/edit?viewport_loc=-192%2C-86%2C2446%2C1152%2C0_0&invitationId=inv_9d92cc1f-bd4c-48a2-bba9-fabb81d63c1f)

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  CONSTRAINT "FK_Medical_histories.patient_id"
+    FOREIGN KEY ("patient_id")
+      REFERENCES "Patients"("id")
+);
+
+CREATE TABLE "Treatment_histories" (
+  "medical_histories_id" int,
+  "treatment_id" int,
+  PRIMARY KEY ("medical_histories_id", "treatment_id")
+);
```
Comment (GedeonTS) [inline schema_based_on_diagram.sql]: - Kindly make sure `medical_histories_id` and `treatment_id` are instead foreign keys, by using `references table_name(id)` to the respective tables.

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: :

Hi Team @Stephen-Adom @ngeelani48 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :tophat:  :+1: 
- Good Documentation :+1: 
- Good PR Description :100: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/* Database schema to keep the structure of entire database. */
+
+CREATE TABLE "Invoices" (
+  "id" bigserial,
+  "total_amount" decimal,
+  "generated_at" timestamp,
+  "payed_at" timestamp,
+  "medical_history_id" int,
```
Comment (pinkmoon25) [inline schema_based_on_diagram.sql:8]: - [ ] [OPTIONAl]  medical_history_id in invoices should also be a foreign key constraint, like 
CONSTRAINT "FK_Invoices.medical_history_id"
    FOREIGN KEY ("medical_history_id")
      REFERENCES medical_histories(id);

**Review by pinkmoon25 — APPROVED** [review]
Hi @ngeelani48 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Gitflow :heavy_check_mark: 
Changes Implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (Stephen-Adom) [reply schema_based_on_diagram.sql:8]: @PinkMoon25 Thank you for the suggestion👍. We will work on it

**Pull request merged**